cmake/CmakeFunctionHelper.cmake

```diff
@@ -93,6 +93,9 @@ set(LOCATION "${SYCLBLAS_GENERATED_SRC}/${blas_level}/${func}/")
 foreach(data ${data_list})
   cpp_type(cpp_data ${data})
   set(container_list "BufferIterator<${cpp_data}>")
+  if(is_dpcpp)
```

Comment:
I think that the generation of the USM routine version should depend on a variable like `SB_ENABLE_USM` rather than `dpcpp` itself.

Moreover, it must be noted that mixing USM and Buffer arguments is possible just in case of header only compilation. Which is fine, but probably it needs to be stated somewhere.

Reply:
This has been addressed in https://github.com/codeplaysoftware/sycl-blas/pull/441